src/libsync/owncloudpropagator.h

```diff
@@ -395,10 +442,15 @@ class OwncloudPropagator : public QObject
         _abortRequested.fetchAndStoreOrdered(true);
         if (_rootJob) {
             // We're possibly already in an item's finished stack
-            QMetaObject::invokeMethod(_rootJob.data(), "abort", Qt::QueuedConnection);
+            // Invoke abort and wait for abort finish
+            // to finalize asynchronously with abortFinished
+            connect(_rootJob.data(), SIGNAL(abortFinished()), this, SLOT(slotAbortFinished()));
+            QTimer::singleShot(jobTimeoutDuration(), this, SLOT(abortTimeout()));
```

Comment:
I give the jobs 5sec to finish on their own asychronously, otherwise I force them to abort synchronously and invoke emitFinished